### ISSUE TITLE
add proper semver support to code base

### ISSUE THREAD

**Issue by bkmartinjr** [chanzuckerberg/cellxgene#1674]
there are several places in the python code where a semver value is interpreted, parsed, compared, etc.  They are all deficient in a number of ways and could be improved by moving to a more full-featured semver package:
* server/converters/cxgtool.py uses a literal string compare and as versions increment, will need more complex logic
* server/cli/upgrade.py is a homebrew comparator, but fails to fully handle pre-release info in comparisons
* server/data_cxg/cxg_adaptor.py uses literal string compares and skips patch level for some generations
etc.

This isn't urgent, but should get sorted out so that various subsystems (eg, schema versioning, CXG versioning) can move forward in a more predictable and controlled manner.

**Comment by signechambers1** [comment]
@maniarathi is this still something we should do? If so, let me know the priority. If not let's close.

**Comment by maniarathi** [comment]
Yeah -- this would still be valuable to do as we increase the number of versioned things in our system. Priority would be P2 imo.

**Comment by roaga** [comment]
I'm looking into this issue now, but it looks like cxgtool.py and cxg_adaptor.py no longer exist and upgrade.py seems to have a lot of semver logic already. Is this issue still a problem, and if so, in which files?
